modules/core/shared/src/main/scala/gem/config/GcalConfig.scala

```diff
@@ -24,27 +25,18 @@ final case class GcalConfig(lamp: GcalLamp, filter: GcalFilter, diffuser: GcalDi
 }
 
 object GcalConfig {
-  // We make this a sealed abstract case class in order to force usage of the
-  // companion object constructor.  The OneAnd head is guaranteed to always be
-  // the minimum GcalArc in the group.
-  sealed abstract case class GcalArcs(arcs: OneAnd[Set, GcalArc]) {
+
+  final case class GcalArcs(arcs: NonEmptySet[GcalArc]) {
     def toList: List[GcalArc] =
-      arcs.head :: arcs.tail.toList
+      arcs.toList
 
-    def toSet: Set[GcalArc] =
-      arcs.tail + arcs.head
+    def toSet: SortedSet[GcalArc] =
+      arcs.toSortedSet
   }
 
   object GcalArcs {
-    /** Constructs GcalArcs such that the GcalArc instances are always in order.
-      */
-    @SuppressWarnings(Array("org.wartremover.warts.OptionPartial"))
-    def apply(arc0: GcalArc, arcs: List[GcalArc]): GcalArcs = {
-      (arc0 :: arcs).toSet.toList.sorted(Order[GcalArc].toOrdering) match {
-        case h :: t => new GcalArcs(OneAnd(h, t.toSet)) {}
-        case Nil    => sys.error("unpossible")
-      }
-    }
+    def of(arc0: GcalArc, arcs: GcalArc*): GcalArcs =
+      GcalArcs(NonEmptySet.of(arc0, arcs: _*))
```

Comment:
Presumably if `arc0` is also a member of `arcs` `NonEmptySet` will do the right thing?

Reply:
Yes, it's basically a newtype over `SortedSet` with constructors guaranteeing it can't be empty.